### PR TITLE
Fix daily task indicator and add sand to daily task indicator

### PR DESCRIPTION
**Pull request by ShaunDreclin**
Also made the plugin check for membership status, completion of diaries/quests, etc. Shouldn't be spamming people that the notifications don't apply to any more, so it can be enabled by default now.

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/VarClientInt.java [hunk 1]
```diff
@@ -36,6 +36,8 @@
 {
 	TOOLTIP_TIMEOUT(1),
 
+	MEMBERSHIP_STATUS(103),
```
Comment (deathbeam) [inline runelite-api/src/main/java/net/runelite/api/VarClientInt.java]: Shouldn't this be memebership days? Or well, shouldn't we use that? Friend list plugin is using it

Comment (Kamielvf) [reply runelite-api/src/main/java/net/runelite/api/VarClientInt.java]: Membership days doesn't get set unless you hit the "Welcome to RuneScape" screen

Comment (deathbeam) [reply runelite-api/src/main/java/net/runelite/api/VarClientInt.java]: Why are we not using that in the friend list then?

Comment (ShaunDreclin) [reply runelite-api/src/main/java/net/runelite/api/VarClientInt.java]: We probably should be, this one is more reliable haha

File: runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksConfig.java [hunk 2]
```diff
@@ -32,38 +34,57 @@
 public interface DailyTasksConfig extends Config
 {
 	@ConfigItem(
+		position = 0,
```
[the remark this answers is not in the document]
Comment (ShaunDreclin) [reply runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksConfig.java]: Not everyone has chat history enabled, I prefer not having history but still showing the daily notices on hopping.

File: runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java [hunk 3]
```diff
@@ -59,105 +62,89 @@
 	@Inject
 	private ChatMessageManager chatMessageManager;
 
-	private boolean hasSentHerbMsg, hasSentStavesMsg, hasSentEssenceMsg, check;
+	private static final int ONE_DAY = 86400;
```
[the remark this answers is not in the document]
Comment (ShaunDreclin) [reply runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java]: done

File: runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java [hunk 4]
```diff
@@ -59,105 +62,89 @@
...
 	@Subscribe
 	public void onGameTick(GameTick event)
 	{
-		if (!check)
-		{
-			return;
-		}
-		
-		check = false;
+		long currentTime = Instant.now().getEpochSecond();
```
[the remark this answers is not in the document]
Comment (ShaunDreclin) [reply runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java]: done

File: runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksConfig.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksConfig.java]: Just enable chat history then. either way this should probably not be part of this PR

[the remark this answers is not in the document]
Comment (ShaunDreclin) [reply runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksConfig.java]: Alright, removed from this one

**Comment by ShaunDreclin** [comment]
Dont merge this yet btw, having an issue getting the notifications to fire after the daily reset if they dont log out for it

**Comment by ShaunDreclin** [comment]
All issues resolved!

screenshot 1375

File: runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java [hunk 5]
```diff
@@ -59,105 +73,107 @@
 	@Inject
 	private ChatMessageManager chatMessageManager;
 
-	private boolean hasSentHerbMsg, hasSentStavesMsg, hasSentEssenceMsg, check;
+	private long lastReset;
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java]: remove this new line

Comment (ShaunDreclin) [reply runelite-client/src/main/java/net/runelite/client/plugins/dailytaskindicators/DailyTasksPlugin.java]: done

**Comment by deathbeam** [comment]
Needs rebase

**Comment by ShaunDreclin** [comment]
rebased

**Pull request merged**